src/components/manager/projects/Project.js

```diff
@@ -65,28 +71,31 @@ const Project = ({
         ...ps,
         ...value,
         data_use: simpleDeepCopy(value.data_use || INITIAL_DATA_USE_VALUE),
+        discovery: newDiscovery || value.discovery,
       }));
     }
-  }, [value]);
+  }, [value, newDiscovery]);
 
-  const handleSave = useCallback(() => {
-    const form = editingForm.current;
-    if (!form) return;
-    form
-      .validateFields()
-      .then((values) => {
-        // Don't save datasets since it's a related set.
-        onSave({
-          identifier: projectState.identifier,
-          title: values.title || projectState.title,
-          description: values.description || projectState.description,
-          data_use: values.data_use || projectState.data_use,
-        });
-      })
-      .catch((err) => {
-        console.error(err);
+  const handleSave = useCallback(async () => {
+    try {
+      const values = await editingForm.validateFields();
+      await onSave({
+        identifier: projectState.identifier,
+        title: values.title || projectState.title,
+        description: values.description || projectState.description,
+        data_use: values.data_use || projectState.data_use,
+        discovery: newDiscovery || values.discovery,
```

Comment:
this line is a bit confusing to me - why does it fall back to values, when everything else falls back to project state? what's the purpose of newDiscovery? (maybe in my PR I forgot to check this part... since it feels like it would have made more sense with the original logic)

Reply:
Line was bad. Thankfully your changes make DropBoxJsonSelect behave like a proper form item, so the hacky hooks for `newDiscovery` are no longer needed here since the form handles everything now.
I tried to do it this way initially but I missed the required props, so your PR is very welcome haha.